src/flask_pyoidc/pyoidc_facade.py

```diff
@@ -50,6 +51,9 @@ def __init__(self, provider_configuration, redirect_uri):
         """
         self._provider_configuration = provider_configuration
         self._client = Client(client_authn_method=CLIENT_AUTHN_METHOD)
+        # Token Introspection is implmented in extension sub-package of the
+        # client in pyoidc
+        self._client_extension = ClientExtension(client_authn_method=CLIENT_AUTHN_METHOD)
```

Comment:
No need for this as a separate variable `oic.extension.client.Client` has allt the same methods as `oic.oic.Client` as far as I can see. So should be possible to just replace `self._client = ClientExtension(...)`.

Reply:
`construct_TokenInstrospectionRequest` and `do_token_introspection` are part of `oic.extension.client`. See https://github.com/OpenIDC/pyoidc/issues/638#issue-428418945

Reply:
You're right and that extension class also contains all the same methods as the base class.
So I'm thinking it should be possible to simply replace `self._client` with the extended class `ClientExtension` instead of adding `self._client_extension` as well.

Reply:
In https://github.com/OpenIDC/pyoidc/issues/638#issue-428418945 , they are discussing to move useful methods from the extension to the base class so we should not replace `_client` with `_client_extension`. Once the migration is complete, I'll remove the `_client_extension` object in future.

Reply:
Now I see your point! 🙂 
However, that issue seems stale so I'd rather do it the other way around: only use `ClientExtension` for now and replace it back when/if the migration is completed in pyoidc. WDYT?

Reply:
Not possible. These are the methods that are only available in `class Client` in `oic.oic` sub-package:

1. `store_registration_info` is fully implemented in base class and is partially implemented in extension class.
2. `construct_AuthorizationRequest` is only available in base class and this method is overridden method of its super class.
3. `verify_id_token` is only available in base class.
4. `do_user_info_request` is only available in base class.

Reply:
Thanks for clarifying - I only skimmed the comparison between the two classes 👍